setup.py

```diff
@@ -29,11 +32,16 @@
     "Programming Language :: Python :: Implementation :: PyPy",
     "Topic :: Software Development :: Libraries :: Python Modules",
 ]
+# mypy plugins are not yet externally distributable, so we must lock tests
```

Comment:
Wait, what!?

This seems to lead the whole thing ad absurdum?

Given this, I would strongly argue for putting the stubs in a separate package – happily in the python-attrs organization that would allow us to release the stubs independently of attrs itself.

Having a hard pin in attrs proper is terrible in so many ways.

Reply:
Yeah, I agree it's pretty lame.   I made [a PR](https://github.com/python/mypy/pull/4734) to make plugins distributable via pip installable entry-points] but the maintainers of mypy feel that its too soon and too much work to pin down a public API for plugins.  I'll put a little more heat on them on the PR to see if we can sneak this in under the radar because I'm afraid the the separate package idea is going to devolve into a nightmare matrix of supported attrs and mypy versions (not to mention that I suspect that the project will be in a regular state of repairing failures introduced by untested changes to attrs), it would be great if you could add your perspective too.

Reply:
>the maintainers of mypy feel that its too soon and too much work to pin down a public API for plugins. I'll put a little more heat on them on the PR to see if we can sneak this in under the radar

The decision to not expose a public API beyond `mypy.api.run` is unlikely to change in the short term. There is too much churn on internal APIs to guarantee a stable public API. I'd like to have a public API, but it is too early. Additionally, we don't have the bandwidth for it at the moment. You can feel free to try to convince Jukka, but I think is pretty firm on his stance as of right now.

>Having a hard pin in attrs proper is terrible in so many ways.

Since you seem against pinning to a mypy release, I'd recommend splitting out to a stub package for now, and merge the stubs back in when we are able to have a stable public API.

>I'm afraid the the separate package idea is going to devolve into a nightmare matrix of supported attrs and mypy versions

I agree this is less than ideal, but you can avoid people shooting themselves in the foot by explicitly marking ranges of versions supported in the stub package setup.py.

Reply:
So just to be clear: this pin is *only* for running *our own* tests and has nothing to do with our users?